File: disturbances/src/main/java/md/vnastasi/cloud/service/DisruptionService.java
package md.vnastasi.cloud.service;

import md.vnastasi.cloud.endpoint.model.Disruption;
import md.vnastasi.cloud.endpoint.model.DisturbanceType;
import reactor.core.publisher.Flux;

import java.util.List;

public interface DisruptionService {

    Flux<Disruption> getDisruptions(List<DisturbanceType> types);
}

File: disruptions/src/main/kotlin/md/vnastasi/cloud/service/DisruptionService.kt
package md.vnastasi.cloud.service

import kotlinx.coroutines.flow.Flow
import md.vnastasi.cloud.endpoint.model.disturbance.Disturbance
import md.vnastasi.cloud.endpoint.model.notification.Notification

interface DisruptionService {

    suspend fun getNotifications(): Flow<Notification>

    suspend fun getDisturbances(): Flow<Disturbance>
}
